canister/src/heartbeat.rs

```diff
@@ -228,6 +232,30 @@ fn maybe_get_successors_request() -> Option<GetSuccessorsRequest> {
     })
 }
 
+#[cfg(target_arch = "wasm32")]
+fn cycles_burn() {
+    let cycles_burnt = ic_cdk::api::cycles_burn(ic_cdk::api::canister_balance128());
+    with_state_mut(|s| {
+        if let Some(metric_cycles_burnt) = &mut s.metrics.cycles_burnt {
+            *metric_cycles_burnt += cycles_burnt;
+        } else {
+            s.metrics.cycles_burnt = Some(cycles_burnt);
+        }
+    });
+}
+
+#[cfg(not(target_arch = "wasm32"))]
+fn cycles_burn() {
+    let cycles_burnt = 1_000_000;
+    with_state_mut(|s| {
+        if let Some(metric_cycles_burnt) = &mut s.metrics.cycles_burnt {
+            *metric_cycles_burnt += cycles_burnt;
+        } else {
+            s.metrics.cycles_burnt = Some(cycles_burnt);
+        }
+    });
```

Comment:
Do we need this logic at all? I don't see anywhere where this is being triggered/used.

Reply:
I am not sure what you are referring to, but if you think about having optional field - we need an optional field because if it is nonoptional deserialization will fail because we have a new field.